src/Microsoft.AspNetCore.Routing/Tree/TreeRouter.cs

```diff
@@ -386,8 +386,7 @@ private VirtualPathData GenerateVirtualPath(VirtualPathContext context, TreeRout
             {
                 if (entry.RequiredLinkValues.ContainsKey(kvp.Key))
                 {
-                    var parameter = entry.Template.Parameters
-                        .FirstOrDefault(p => string.Equals(p.Name, kvp.Key, StringComparison.OrdinalIgnoreCase));
+                    var parameter = entry.Template.GetParameter(kvp.Key);
```

Comment:
`context.Values` is a `RVD` which has a struct enumerator already. I don't think it's causing trouble.